scripts/apitypings/main.go

```diff
@@ -253,10 +296,62 @@ func (g *Generator) posLine(obj types.Object) string {
 }
 
 // buildStruct just prints the typescript def for a type.
-func (g *Generator) buildStruct(obj types.Object, st *types.Struct) (string, error) {
+func (g *Generator) buildUnion(obj types.Object, st *types.Union) (string, error) {
 	var s strings.Builder
 	_, _ = s.WriteString(g.posLine(obj))
-	_, _ = s.WriteString(fmt.Sprintf("export interface %s ", obj.Name()))
+
+	allTypes := make([]string, 0, st.Len())
+	var optional bool
+	for i := 0; i < st.Len(); i++ {
+		term := st.Term(i)
+		scriptType, err := g.typescriptType(term.Type())
+		if err != nil {
+			return "", xerrors.Errorf("union %q for %q failed to get type: %w", st.String(), obj.Name(), err)
+		}
+		allTypes = append(allTypes, scriptType.ValueType)
+		optional = optional || scriptType.Optional
+	}
+
+	qMark := ""
+	if optional {
+		qMark = "?"
+	}
+
+	s.WriteString(fmt.Sprintf("export type %s%s = %s\n", obj.Name(), qMark, strings.Join(allTypes, " | ")))
```

Comment:
I don't think question marks are supported on types, only on interface fields.

Reply:
I have read the CLA Document and I hereby sign the CLA

Reply:
Good catch. Made it nullable